staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -166,3 +166,51 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityAndFairnessConfiguration holds API Priority and Fairness parameters.
+type PriorityAndFairnessConfiguration struct {
+	metav1.TypeMeta
+
+	// WorkEstimatorConfig holds work estimator parameters.
+	WorkEstimatorConfiguration WorkEstimatorConfiguration
+}
+
+// WorkEstimatorConfiguration holds work estimator parameters.
+type WorkEstimatorConfiguration struct {
+	// ListWorkEstimatorConfiguration holds work estimator parameters related to list requests.
+	ListWorkEstimatorConfiguration ListWorkEstimatorConfiguration
+	// MutatingWorkEstimatorConfiguration holds work estimator
+	// parameters related to watches of mutating objects.
+	MutatingWorkEstimatorConfiguration MutatingWorkEstimatorConfiguration
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	//
+	// NOTE: work_estimate_seats_samples metric uses the value of maximumSeats
+	// as the upper bound, so when we change maximumSeats we should also
+	// update the buckets of the metric.
+	MaximumSeats uint64
+}
+
+// ListWorkEstimatorConfiguration holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfiguration struct {
+	// ObjectsPerSeat is a number of objects from a single list request per seat.
+	ObjectsPerSeat float64
+}
+
+// MutatingWorkEstimatorConfiguration holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfiguration struct {
+	// Enabled enables mutating work estimator.
+	// This option will be removed once we have better proof from
+	// production usecases that the feature works as expected.
+	Enabled bool
```

Comment:
At first glance, this looks like like server-side config that could dramatically change the behavior of the same AP&F API objects. That doesn't seem like a great thing for portable P&F config. If this is an "escape-hatch" type option while the feature is in development, I don't think API server config is the right place for it... a feature gate we'd expect to progress and be removed at GA seems more correct

Reply:
Fair enough, I'll revert the change that extracts this `Enabled` field. I'll leave it as `const` in the code for now.

Reply:
Removed.

Reply:
Hey, this was discussed earlier (https://github.com/kubernetes/kubernetes/pull/111222#discussion_r927347533) and the opposite conclusion reached.